client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,789 @@
+import axe from 'axe-core';
+
+import { Controller } from '@hotwired/stimulus';
+import type { ContextObject } from 'axe-core';
+import {
+  getAxeConfiguration,
+  getA11yReport,
+  renderA11yResults,
+} from '../includes/a11y-result';
+import { wagtailPreviewPlugin } from '../includes/previewPlugin';
+import {
+  getPreviewContentMetrics,
+  renderContentMetrics,
+} from '../includes/contentMetrics';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+import { setOptionalInterval } from '../utils/interval';
+
+const runContentChecks = async () => {
+  axe.registerPlugin(wagtailPreviewPlugin);
+
+  const contentMetrics = await getPreviewContentMetrics({
+    targetElement: 'main, [role="main"], body',
+  });
+
+  // This requires Wagtail's preview plugin for axe to be registered in the
+  // preview iframe, which is not done in tests as the registration happens via
+  // the userbar.
+  if (!contentMetrics) return;
```

Comment:
Aside from adding this and the other null/undefined checks, TypeScript annotations, and updating the `#preview-iframe` to `#w-preview-iframe` selector, I believe I didn't change anything to these two functions.

Reply:
OK, I think when this is pulled out to other code we would probably pass in the DOM element via the event being fired or something similar.